### PR TITLE
[SPARK-11259] [ML] Params.validateParams() should be called automatically

### PR DESCRIPTION
**Pull request by yanboliang**
See JIRA: https://issues.apache.org/jira/browse/SPARK-11259

### CODE REVIEW

**Comment by jkbradley** [comment]
This looks fine to me, but would you mind going ahead and adding a call to validateParams in all transformSchema implementations?

File: mllib/src/main/scala/org/apache/spark/ml/Predictor.scala [hunk 1]
```diff
@@ -113,6 +113,7 @@ abstract class Predictor[
   private[ml] def featuresDataType: DataType = new VectorUDT
 
   override def transformSchema(schema: StructType): StructType = {
+    validateParams()
```
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/ml/Predictor.scala]: In cases like this where the Estimator and Model share validateAndTransformSchema, the call could be moved to validateAndTransformSchema.

**Comment by yanboliang** [comment]
Jenkins, test this please.

**Comment by yanboliang** [comment]
Jenkins, test this please.

**Comment by yanboliang** [comment]
@jkbradley

**Comment by jkbradley** [comment]
Apologies, but I may need to hold off on review for a bit as we prepare the 1.6 release.  Once that's out of the way, I'll be able to resume regular code review.  Thanks for being patient!

**Comment by jkbradley** [comment]
I can do a final pass after merge conflicts are resolved.  Sorry for the delay!

**Comment by yanboliang** [comment]
Jenkins, test this please.

**Comment by jkbradley** [comment]
Thanks for the updates!  The only issue I spotted is that LDA should be updated too (now that it is in spark.ml).

**Comment by jkbradley** [comment]
@yanboliang Thanks for the update.  LGTM
Merging with master

**Pull request closed without merging**